Automattic/woocommerce-payments
# Prefill the Business URL on the onboarding form

Fixes #6488 

#### Changes proposed in this Pull Request



* Add the ability for the `OnboardingContextProvider` to support initial values.
* Prefill the business URL so that the initial value on the form will be the users site URL.
* Update the tests to ensure the initial value fill works as expected.







#### Testing instructions

Note: I recommend not completing the final step of the onboarding while testing, since this will create the Stripe account and you'll have to delete it to re-test.

* With Progressive Onboarding enabled, start a new onboarding
* Verify that the URL of the site in the "business URL" step matches your local website (if it is `locahost`, it should say `https://wcpay.test`)
* Verify that a validation error does not occur when leaving the value untouched and clicking continue on the form (after filling out the other needed values).
* Verify that emptying the field causes a validation error.
* Verify that the value can be changed to another value (e.g. `http://google.com`) and it will be accepted.

Bonus: Spin up a new JN site using [live branches](https://github.com/Automattic/woocommerce-payments/pull/6579#issuecomment-1604195776), install the dev tools and enable `Progressive Onboarding` and ensure the URL populates correctly on the PO form.






-------------------


**Post merge**
